photon-lib/src/main/scala/com/linkedin/photon/ml/hyperparameter/search/RandomSearch.scala

```diff
@@ -68,65 +64,45 @@ class RandomSearch[T](
 
     require(n > 0, "The number of results must be greater than zero.")
 
-    // Load the initial observations
-    observations.init.foreach { case (candidate, value) =>
-      onObservation(candidate, value)
-    }
-
-    // Load the prior data observations
-    priorObservations.foreach { case (candidate, value) =>
-      onPriorObservation(candidate, value)
-    }
-
-    val (results, _) = (0 until n).foldLeft((List.empty[T], observations.last)) {
-      case ((models, (lastCandidate, lastObservation)), _) =>
+    if (observations.isEmpty) {
 
-        val candidate = next(lastCandidate, lastObservation)
+      val candidate = drawCandidates(1)(0, ::).t
 
-        // Discretize values specified as discrete
-        discreteParams.foreach { index =>
-          candidate(index) = round(candidate(index))
-        }
+      // Discretize values specified as discrete
+      discretizeCandidate(candidate, discreteParams)
 
-        val (observation, model) = evaluationFunction(candidate)
+      val (_, model) = evaluationFunction(candidate)
 
-        (models :+ model, (candidate, observation))
+      Seq(model) ++ (if (n == 1) Seq() else findWithPriors(n - 1, convertObservations(Seq(model)), priorObservations))
     }
 
-    results
-  }
+    else {
```

Comment:
Lines 77 - 79 should be one line: `} else {`

Any reason that these two functions were merged into one? IMO it's more confusing to follow

Reply:
I guess in the previous version, the reason to keep four "find" functions is for four cases with `observations` and `priorObservations` be present or not. Therefore, if statement is needed to decide which function to be called. So I prefer to use a general function to do all the jobs. I am open to change it back. Thanks for your comments.

Reply:
IMO it would be better for readability if we split this into 2 functions rather than one big function bifurcated by an if/else branch.

Reply:
If the argument `observations` in the previous `priorWithPrior` is empty, it will cause an error. We didn't do the check for that. That's why I add an `if` statement. I agree on splitting back to the original four functions but suggest keep this if-else statement.